### DISCUSSION TITLE
Benefits / Use Cases of Queue Messages ->On Failure vs Always

### DISCUSSION THREAD

**Discussion by JeffBenedict** [nextgenhealthcare/connect#5739] [Frequently Asked Questions]
In MC 3.x +, under Destination Settings/Queue Messages;  I read and understand the definition of On Failure vs Always. We have ours primary set to On Failure. Can someone explain use case as to when / why we'd want to use "always" ?    Thanks. 

![image](https://user-images.githubusercontent.com/106398018/228598274-f6c0ee29-f249-473c-8466-11752dff2d02.png)

**Comment by jonbartels** [comment]
Always is truly asynchronous. It will queue every single message.

Queue On Failure is only asynchronous *when there is an error*. So successful messages will not queue and send synchronously. Failed messages will queue.

IMO "Always" is a better default choice because the behavior is consistent for every message.
